internal/formats/common/cyclonedxhelpers/format.go

```diff
@@ -0,0 +1,90 @@
+package cyclonedxhelpers
+
+import (
+	"time"
+
+	"github.com/CycloneDX/cyclonedx-go"
+	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/internal/version"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/anchore/syft/syft/source"
+	"github.com/google/uuid"
+)
+
+func ToFormatModel(s sbom.SBOM) *cyclonedx.BOM {
+	cdxBOM := cyclonedx.NewBOM()
+	versionInfo := version.FromBuild()
+
+	cdxBOM.SerialNumber = uuid.New().URN()
+	cdxBOM.Metadata = toBomDescriptor(internal.ApplicationName, versionInfo.Version, s.Source)
+
+	packages := s.Artifacts.PackageCatalog.Sorted()
+	components := make([]cyclonedx.Component, len(packages))
+	for i, p := range packages {
+		components[i] = toComponent(p)
+	}
+	cdxBOM.Components = &components
+
+	return cdxBOM
+}
+
+// NewBomDescriptor returns a new BomDescriptor tailored for the current time and "syft" tool details.
+func toBomDescriptor(name, version string, srcMetadata source.Metadata) *cyclonedx.Metadata {
+	return &cyclonedx.Metadata{
+		Timestamp: time.Now().Format(time.RFC3339),
+		Tools: &[]cyclonedx.Tool{
+			{
+				Vendor:  "anchore",
+				Name:    name,
+				Version: version,
+			},
+		},
+		Component: toBomDescriptorComponent(srcMetadata),
+	}
+}
+
+func toComponent(p pkg.Package) cyclonedx.Component {
+	return cyclonedx.Component{
+		Type:       cyclonedx.ComponentTypeLibrary,
+		Name:       p.Name,
+		Version:    p.Version,
+		PackageURL: p.PURL,
+		Licenses:   toLicenses(p.Licenses),
+	}
+}
+
+func toBomDescriptorComponent(srcMetadata source.Metadata) *cyclonedx.Component {
+	switch srcMetadata.Scheme {
+	case source.ImageScheme:
+		return &cyclonedx.Component{
+			Type:    cyclonedx.ComponentTypeContainer,
+			Name:    srcMetadata.ImageMetadata.UserInput,
+			Version: srcMetadata.ImageMetadata.ManifestDigest,
+		}
+	case source.DirectoryScheme:
+		return &cyclonedx.Component{
+			Type: cyclonedx.ComponentTypeFile,
+			Name: srcMetadata.Path,
+		}
+	}
+
+	return nil
+}
+
+func toLicenses(ls []string) *cyclonedx.Licenses {
+	if len(ls) == 0 {
+		return nil
+	}
+
+	lc := make(cyclonedx.Licenses, len(ls))
```

Comment:
Same comment here, index assignment vs appened. When do we use one vs the other? I know we use both in this codebase so clarity on the differences and choice would help when doing PR review.

Reply:
`<reflection on the meta question about reviews>`

I think both are OK and valid in different circumstances (my take is that this is in the "up to the PR author" thing, but not out of bounds for review comment... lots of space for "depending on the context"). Technically via assignment has fewer allocations, but again, I'm a +0 on that concern. 

More 2 🪙 ... from a review point of view, we are trying to prefer code in that is:
1. harder to abuse / misuse 
2. resistant to future problems / brittleness 
3. easy to read / grok / understand

These are all mostly subjective, but are more clear when comparing implementation alternatives.

Take for instance incorporating @spiffcs suggestion:

```
func toLicenses(ls []string) *cyclonedx.Licenses {
	var lc cyclonedx.Licenses
	for _, licenseName := range ls {
		lc = append(lc, cyclonedx.LicenseChoice{
			License: &cyclonedx.License{
				Name: licenseName,
			},
		})
	}

	return &lc
}
```

The upside about this is that the implementation is shorter without increasing the "comprehensional density" (warning: made up phrase)... that is, I don't need to "unpack" extra hidden behavior from this alternative implementation. But it expresses the same behavior in a smaller space with fewer concerns (e.g. no need for a guard clause, better default nil behavior, one less variable to worry about).

That all being said, I personally consider comments about these kinds of things to be in the `nit` category. What I specifically mean by a "nit" comment is a "take it or leave it, optional comment" aimed solely at potential improvements to the code but do not block the approval process --it's up to the PR author to moderate the comment (incorporate or not).

`</reflection>`